src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -302,9 +302,12 @@ protected void install() {
                 });
 
         Topics strategyTopics = config.lookupTopics(CONFIGURATION_CONFIG_KEY, CONFIGURATION_STRATEGY_TOPIC);
+        final Strategy[] currentStrategy = { DEFAULT_STRATEGY };
```

Comment:
Why have an array? Can't we just an `AtomicReference`?

Reply:
The reference changes though because it creates a new strategy object, even though they are logically equal

so you can't use `compareAndSet` because it is doing `==`. Then it's no different - you have to do `!ar.get().equals(strategy)` and it's functionally no different.

What's wrong with the array approach?

Reply:
I just think it looks a little cleaner rather than with an array.